src/Resources/contao/library/Contao/Controller.php

```diff
@@ -1003,6 +1003,7 @@ public static function redirect($strLocation, $intStatus=303)
 			exit;
 		}
 
+		$strLocation = ltrim($strLocation, '/');
```

Comment:
That only makes sense if the automated fixing could generate false positives. But in this case it cannot. 
Your proposal should look like this

```php
if ($strLocation[0] == '/')
{
    throw new InvalidArgumentException('The target location string contains a leading slash which it must not!');
}
```

**THAT** would be detecting wrong input. Right now it just does nothing. It passes it on and generates wrong URL's. Either you fix it automatically (which you can do in this case) or you have to throw an Exception :-)